core/src/main/java/com/linecorp/armeria/client/HttpClientPipelineConfigurator.java

```diff
@@ -729,18 +729,9 @@ private Http2ClientConnectionHandler newHttp2ConnectionHandler(Channel ch, Sessi
 
         final Http2ClientConnectionHandlerBuilder builder =
                 new Http2ClientConnectionHandlerBuilder(ch, clientFactory, protocol);
-        builder.codec(decoder, encoder)
-               .initialSettings(http2Settings());
-
-        final long timeout = clientFactory.idleTimeoutMillis();
-        if (timeout > 0) {
-            builder.gracefulShutdownTimeoutMillis(timeout);
```

Comment:
The previous default graceful shutdown timeout was 10 seconds. However, this could lead to sockets on the server-side entering a time-wait state. Given that the server-side graceful shutdown is set to 1 second, I believe we should align the timeout accordingly. Therefore, I've chosen to set it to 1 second.
cc @line/dx

Reply:
1 second on the server side sounds too short to support graceful shutdown.

A new inflight request could be received as the connection drain starts. The request will fail if it is not finished in 1 second. What do you think of reverting the following change and setting the default request timeout to `gracefulShutdownTimeoutMillis ()`?
https://github.com/line/armeria/blob/1d268aa278524997ad5c1cc8f66661cfabbb6dd1/core/src/main/java/com/linecorp/armeria/server/Http2ServerConnectionHandlerBuilder.java#L43-L45

The default response timeout may be set as the default graceful timeout for the client side.

Reply:
That's a good point. However, I'm a bit worried having two separate graceful shutdown mechanisms on the server-side. 
> The request will fail if it is not finished in 1 second. 

If you worried about this situation, how about just changing the default value of server connection drain duration?

Reply:
> how about just changing the default value of server connection drain duration?

It is acceptable if we can meet the specifications below:
https://github.com/line/armeria/blob/1d268aa278524997ad5c1cc8f66661cfabbb6dd1/core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java#L772-L778

Currently, a connection seems closed without waiting for the active streams to be closed.

Reply:
> new requests will be ignored

This part only applies to HTTP/1, so the comment needs to be modified.

Reply:
If `responseEncoder.keepAliveHandler().needsDisconnection() == true` by `MaxConnectionAgeExceededTask`, the specification could be violated.
The first response after `needsDisconnection == true` calls `channel.close()` and then the connection may be closed in 1 second while other responses are being processed.

Reply:
I had a discussion with @ikhoon, and it seems that the topic wasn't related to the features of this pull request, but rather about a feature that we still need to implement. I'll address it in the next pull request.

Reply:
Oh, it seems like previous implementation had been sharing `idleTimeout` for both keepAlive and graceful shutdown features 🤔

Reply:
That is correct. 👍